File: go/rotational-cipher/rotational_cipher.go
// Package rotationalcipher implements a simple library for rotational ceasar cipher
package rotationalcipher

import (
	"unicode"
)

// RotationalCipher given plain text and shift key
// shifts all the letters and returns cipher text
func RotationalCipher(text string, shift int) string {
	shifted := make([]rune, len(text))
	for i, r := range text {
		shifted[i] = rotate(r, shift)
	}

	return string(shifted)
}

func rotate(r rune, shift int) rune {
	if !unicode.IsLetter(r) {
		return r
	}

	offset := int('a')
	if unicode.IsUpper(r) {
		offset = int('A')
	}

	s := rune(((int(r)+shift)-offset)%26 + offset)
	return s
}

File: csharp/rotational-cipher/RotationalCipher.cs
using System;
using System.Text;

public static class RotationalCipher
{
    public static string Rotate(string text, int shiftKey)
    {
        var shifted = new StringBuilder();
        foreach (var ch in text)
        {
            shifted.Append(Rotate(ch, shiftKey));
        }

        return shifted.ToString();
    }

    private static char Rotate(char character, int shiftKey)
    {
        if (!char.IsLetter(character))
        {
            return character;
        }

        var offset = char.IsUpper(character) ? 'A' : 'a';
        return (char)(((character + shiftKey) - offset) % 26 + offset);
    }
}